drivers/serial/Kconfig

```diff
@@ -46,16 +36,10 @@ config NRF_SW_LPUART_DEFAULT_TX_TIMEOUT
 	  Timeout is used in uart_poll_out and uart_fifo_fill (if interrupt
 	  driven API is enabled).
 
-config NRF_SW_LPUART_INT_DRIVEN
```

Comment:
I think that's a bit misleading, specially because this is not how upstream UART drivers work to my understanding. Here we have

```
The low power UART driver implements the standard *asynchronous UART API* that can be enabled with the :kconfig:option:`CONFIG_UART_ASYNC_API` configuration option.
Alternatively, you can also enable the *interrupt-driven UART API* using the :kconfig:option:`CONFIG_NRF_SW_LPUART_INT_DRIVEN` configuration option.
```

As far as I can tell they don't seem to be mutually exclusive options in the upstream API header.

Reply:
@nordic-krch ping

Reply:
they are mutually exclusive since when interrupt driven mode is used then async callback is handled internally thus not available for the user and rx is enabled at initialization.

Reply:
I still don't understand why this option i removed. Let say that I have 2 uarts in the system. uart0 i want to use with lpuart with async api and uart1 instance needs to use interrupt driven API. I need to enable both API's and configure uart0 to use async and uart1 to use interrupt driven. Without this PR lpuart will be forced to use interrupt driven API, right?